crates/anvil/tests/it/eip4844.rs

```diff
@@ -204,3 +204,43 @@ async fn can_check_blob_fields_on_genesis() {
     assert_eq!(block.header.blob_gas_used, Some(0));
     assert_eq!(block.header.excess_blob_gas, Some(0));
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn can_correctly_estimate_blob_gas() {
+    let node_config = NodeConfig::test().with_hardfork(Some(EthereumHardfork::Cancun.into()));
+    let (_api, handle) = spawn(node_config).await;
+
+    let provider = http_provider(&handle.http_endpoint());
+
+    let accounts = provider.get_accounts().await.unwrap();
+    let bob = accounts[1];
+
+    let sidecar: SidecarBuilder<SimpleCoder> = SidecarBuilder::from_slice(b"Blobs are fun!");
+    let sidecar = sidecar.build().unwrap();
+
+    let gas_price = provider.get_gas_price().await.unwrap();
+    let tx = TransactionRequest::default()
+        .with_to(bob)
+        .with_max_fee_per_blob_gas(gas_price) // TODO: this should not be required
```

Comment:
Added a basic test, confirming the behavior reported - doesn't look like the proposed fix addresses it yet, will require some further digging into whether it is on Alloy's side or Anvil's side

Reply:
Confirming that this is an Anvil issue, was able to send a transaction without specifying the `gas_price` to a local Reth node w/ Alloy bindings